src/ape/managers/chain.py

```diff
@@ -0,0 +1,143 @@
+from typing import List, Optional
+
+from dataclassy import dataclass
+
+from ape.api import BlockAPI, ProviderAPI, TestProviderAPI
+from ape.exceptions import ChainError, ProviderNotConnectedError, UnknownSnapshotError
+from ape.types import SnapshotID
+
+from .networks import NetworkManager
+
+
+@dataclass
+class BlockContainer:
+    """
+    A list of blocks on the chain.
+
+    Usages example::
+
+        from ape import chain
+
+        latest_block = chain.blocks[-1
+
+    """
+
+    _provider: ProviderAPI
+
+    def __getitem__(self, block_number: int) -> BlockAPI:
+        """
+        Get a block by number.
```

Comment:
Probably explain the behavior of a negative number here (e.g. -1 is head/chain tip, -2 is one behind, etc.)